### PR TITLE
added metamechs directory, created timberwolf keyboard & moved daishi to it

### PR DESCRIPTION
**Pull request by Croktopus**
## Description

MetaMechs is the company name under which I'm designing keyboards, so I created a directory for current and future projects, moved an old project (daishi)into it, and added the keyboard I'll be shipping soon (timber wolf).

Firmware is pretty much done on the Timber Wolf, I just need to finish setting up QMK Configurator and maybe VIA (I'm sure I'll be posting many questions in the discord in the coming days).

## Types of Changes

- [x] Keyboard (addition or update)

## Checklist

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/metamechs/timberwolf/keymaps/prime/keymap.c [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+		_______,_______,_______,_______                        ,_______                ,_______,_______,_______                ,_______,_______,_______  \
+    ),
+    [3] = LAYOUT(
+		_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______        ,_______,_______,_______, \
+		_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______, \
+		_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______        ,_______,_______, \
+		_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______        ,_______,_______, \
+		_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______        ,_______,_______, \
+		_______,_______,_______,_______                        ,_______                ,_______,_______,_______                ,_______,_______,_______  \
```
Comment (fauxpark) [inline keyboards/metamechs/timberwolf/keymaps/prime/keymap.c]:
```suggestion
		KC_MUTE,KC_ESC ,KC_F1  ,KC_F2  ,KC_F3  ,KC_F4  ,KC_F5  ,KC_F6  ,KC_F7  ,KC_F8  ,KC_F9  ,KC_F10 ,KC_F11 ,KC_F12         ,KC_INS ,KC_HOME,KC_PGUP,
		MO(1)  ,KC_GRV ,KC_1   ,KC_2   ,KC_3   ,KC_4   ,KC_5   ,KC_6   ,KC_7   ,KC_8   ,KC_9   ,KC_0   ,KC_MINS,KC_EQL ,KC_BSPC,KC_GRV ,KC_END ,KC_PGDN,
		MO(1)  ,KC_TAB ,KC_Q   ,KC_W   ,KC_E   ,KC_R   ,KC_T   ,KC_Y   ,KC_U   ,KC_I   ,KC_O   ,KC_P   ,KC_LBRC,KC_RBRC,KC_BSLS        ,KC_DEL ,KC_PSCR,
		MO(1)  ,KC_CAPS,KC_A   ,KC_S   ,KC_D   ,KC_F   ,KC_G   ,KC_H   ,KC_J   ,KC_K   ,KC_L   ,KC_SCLN,KC_QUOT,KC_NUHS,KC_ENT         ,KC_SLCK,KC_PENT,
		MO(1)  ,KC_LSFT,KC_NUBS,KC_Z   ,KC_X   ,KC_C   ,KC_V   ,KC_B   ,KC_N   ,KC_M   ,KC_COMM,KC_DOT ,KC_SLSH,KC_RSFT,MO(1)          ,KC_UP  ,KC_PAUS,
		MO(1)  ,KC_LCTL,KC_LGUI,KC_LALT                        ,KC_SPC                 ,KC_LALT,KC_RGUI,KC_RCTL                ,KC_LEFT,KC_DOWN,KC_RGHT
    ),
    [1] = LAYOUT(
		RESET  ,RESET  ,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______        ,_______,_______,_______,
		_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,
		_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______        ,_______,_______,
		_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______        ,_______,_______,
		_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______        ,_______,_______,
		_______,_______,_______,_______                        ,_______                ,_______,_______,_______                ,_______,_______,_______
```

File: keyboards/metamechs/timberwolf/keymaps/via/keymap.c [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+		_______,_______,_______,_______                        ,_______                ,_______,_______,_______                ,_______,_______,_______  \
+    ),
+    [3] = LAYOUT(
+		_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______        ,_______,_______,_______, \
+		_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______, \
+		_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______        ,_______,_______, \
+		_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______        ,_______,_______, \
+		_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______        ,_______,_______, \
+		_______,_______,_______,_______                        ,_______                ,_______,_______,_______                ,_______,_______,_______  \
```
Comment (fauxpark) [inline keyboards/metamechs/timberwolf/keymaps/via/keymap.c]:
```suggestion
		KC_MUTE,KC_ESC ,KC_F1  ,KC_F2  ,KC_F3  ,KC_F4  ,KC_F5  ,KC_F6  ,KC_F7  ,KC_F8  ,KC_F9  ,KC_F10 ,KC_F11 ,KC_F12         ,KC_INS ,KC_HOME,KC_PGUP,
		MO(1)  ,KC_GRV ,KC_1   ,KC_2   ,KC_3   ,KC_4   ,KC_5   ,KC_6   ,KC_7   ,KC_8   ,KC_9   ,KC_0   ,KC_MINS,KC_EQL ,KC_BSPC,KC_GRV ,KC_END ,KC_PGDN,
		MO(1)  ,KC_TAB ,KC_Q   ,KC_W   ,KC_E   ,KC_R   ,KC_T   ,KC_Y   ,KC_U   ,KC_I   ,KC_O   ,KC_P   ,KC_LBRC,KC_RBRC,KC_BSLS        ,KC_DEL ,KC_PSCR,
		MO(1)  ,KC_CAPS,KC_A   ,KC_S   ,KC_D   ,KC_F   ,KC_G   ,KC_H   ,KC_J   ,KC_K   ,KC_L   ,KC_SCLN,KC_QUOT,KC_NUHS,KC_ENT         ,KC_SLCK,KC_PENT,
		MO(1)  ,KC_LSFT,KC_NUBS,KC_Z   ,KC_X   ,KC_C   ,KC_V   ,KC_B   ,KC_N   ,KC_M   ,KC_COMM,KC_DOT ,KC_SLSH,KC_RSFT,MO(1)          ,KC_UP  ,KC_PAUS,
		MO(1)  ,KC_LCTL,KC_LGUI,KC_LALT                        ,KC_SPC                 ,KC_LALT,KC_RGUI,KC_RCTL                ,KC_LEFT,KC_DOWN,KC_RGHT
    ),
    [1] = LAYOUT(
		RESET  ,RESET  ,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______        ,_______,_______,_______,
		_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,
		_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______        ,_______,_______,
		_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______        ,_______,_______,
		_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______        ,_______,_______,
		_______,_______,_______,_______                        ,_______                ,_______,_______,_______                ,_______,_______,_______
    ),
    [2] = LAYOUT(
		_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______        ,_______,_______,_______,
		_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,
		_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______        ,_______,_______,
		_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______        ,_______,_______,
		_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______        ,_______,_______,
		_______,_______,_______,_______                        ,_______                ,_______,_______,_______                ,_______,_______,_______
    ),
    [3] = LAYOUT(
		_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______        ,_______,_______,_______,
		_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,
		_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______        ,_______,_______,
		_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______        ,_______,_______,
		_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______        ,_______,_______,
		_______,_______,_______,_______                        ,_______                ,_______,_______,_______                ,_______,_______,_______
```

**Comment by Croktopus** [comment]
> `timberwolf` needs a default keymap, however the other keymaps look rather inconsistent. They build, but given the order provided to the LAYOUT macros, I doubt they will function correctly.

one of the weird things about this keyboard is there are 3 physical layouts supported, each with their own top case, but the same pcb. i've added a default keymap, but it may make more sense to split them up into separate keyboards - I was just hoping to avoid that as it would it so any time I wanted to make changes to the firmware, I'd have to do it 3 times.

The physical PCB wiring is weird - I had to do some funky things to get everything to fit like I wanted. Wouldn't make the same choices if I was designing it today, but it seems to work.

**Comment by Croktopus** [comment]
> The Configurator support here is a mess, but I'm thinking it may be easier to just merge this in and I can fix/extend the Configurator support in another PR. Given the way the keyboard works (in the hardware assembly sense), we should start with a `LAYOUT_all` macro (rename all instances of `LAYOUT` to `LAYOUT_all`), then add macros for the A, B, and Prime assemblies.

Yeah I've never worked with Configurator before so the original intent was to simply get it in and then work from there, I can try making some layout macros tonight too - those are just separate layouts in the keyboard.h file, correct (so LAYOUT_all, LAYOUT_prime, LAYOUT_a, LAYOUT_b, and then maybe some extras for stuff like ISO for example)?

**Comment by noroadsleft** [comment]
> Yeah I've never worked with Configurator before so the original intent was to simply get it in and then work from there, I can try making some layout macros tonight too - those are just separate layouts in the keyboard.h file, correct (so LAYOUT_all, LAYOUT_prime, LAYOUT_a, LAYOUT_b, and then maybe some extras for stuff like ISO for example)?

Yeah. I'd suggest the following configuration:

- `LAYOUT_all` (every possible location has a switch)
- `LAYOUT_prime_ansi`/`LAYOUT_prime_iso` – ANSI and ISO variants for the Prime
- `LAYOUT_a_ansi`/`LAYOUT_a_iso` – ANSI and ISO variants for the A
- `LAYOUT_b_ansi`/`LAYOUT_b_iso` – ANSI and ISO variants for the B

Also, any locations where the user has the choice of 1x2U vertical vs. 2x1U should be arranged as 2x1U, should a user wish to split any or all of those position in two.

Note that QMK custom is that Enter key is on the home row for ISO layouts macros:

```
... KC_SCLN, KC_QUOT, KC_NUHS, KC_ENT, ...
```
:+1: ...

```
... KC_P,    KC_LBRC, KC_RBRC, KC_ENT, ...
```
:-1:

**Comment by Croktopus** [comment]
ended up entirely remaking the pcb, split off prime, a, and b into their own directories, each now has its own default, ansi, and iso layout (via is placeholder, will add via support in coming days/weeks), tested them and they seem good. configurator info.json files im less sure about but dont know how to troubleshoot that without merging.

File: keyboards/metamechs/timberwolf/timberwolf.c [hunk 3]
```diff
@@ -16,11 +16,12 @@
...
-        writePin(E2, led_state.caps_lock);
-    }
-    return res;
-}
+    bool runDefault = led_update_user(led_state);
+    if (led_state.caps_lock == 0) {
+      backlight_disable();
+    } else if (led_state.caps_lock == 1) {
+		backlight_enable();
+	}
```
[the remark this answers is not in the document]
Comment (Croktopus) [reply keyboards/metamechs/timberwolf/timberwolf.c:25]: man i was so proud of myself for coming up with some code all on my own too. and its just not quite right lmao. fixed, thanks

[the remark this answers is not in the document]
Comment (fauxpark) [reply keyboards/metamechs/timberwolf/timberwolf.c]: Well, it works just as well, but this is a slightly neater way to do it. It probably compiles into roughly the same thing anyway :P

**Comment by noroadsleft** [comment]
Tagged as a Breaking Change due to the Daishi moving.

I'll give this a full review tomorrow. It's late and I'm tired.

**Comment by Croktopus** [comment]
alright ill close this today and submit a new PR with just timber wolf

**Pull request closed without merging**